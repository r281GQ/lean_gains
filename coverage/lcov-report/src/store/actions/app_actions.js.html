<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/store/actions/app_actions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/store/actions</a> app_actions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>21/21</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>21/21</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22</td><td class="line-coverage quiet"><span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const SELECT_WORKOUT_LOG_MONTH = 'SELECT_WORKOUT_LOG_MONTH';
export const OPEN_CONFIRM_DELETE_MODAL = 'OPEN_CONFIRM_DELETE_MODAL';
export const CLOSE_CONFIRM_DELETE_MODAL = 'CLOSE_CONFIRM_DELETE_MODAL';
export const SET_SELECTED_WORKOUT_LOG = 'SET_SELECTED_WORKOUT_LOG';
export const OPEN_SIDE_BAR = 'OPEN_SIDE_BAR';
export const CLOSE_SIDE_BAR = 'CLOSE_SIDE_BAR';
export const SELECT_DAILY_LOG_MONTH = 'SELECT_DAILY_LOG_MONTH';
export const SET_SELECTED_WORKOUT_TARGET = 'SET_SELECTED_WORKOUT_TARGET';
export const SET_SELECTED_DAILY_LOG = 'SET_SELECTED_DAILY_LOG';
export const INIT_API = 'INIT_API';
export const CLOSE_API = 'CLOSE_API';
export const OPEN_MESSAGE_BAR = 'OPEN_MESSAGE_BAR';
export const CLOSE_MESSAGE_BAR = 'CLOSE_MESSAGE_BAR';
export const SET_MESSAGE = 'SET_MESSAGE';
export const INIT_FETCH = 'INIT_FETCH';
export const CLOSE_FETCH = 'CLOSE_FETCH';
export const SET_CALORIE_LOG_DAY = 'SET_CALORIE_LOG_DAY';
export const OPEN_CONSENT_MODAL = 'OPEN_CONSENT_MODAL';
export const CLOSE_CONSENT_MODAL = 'CLOSE_CONSENT_MODAL';
export const ADD_CONSENT = 'ADD_CONSENT';
export const REMOVE_CONSENT = 'REMOVE_CONSENT';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
