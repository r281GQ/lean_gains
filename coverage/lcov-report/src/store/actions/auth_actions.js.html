<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/store/actions/auth_actions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/store/actions</a> auth_actions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const INIT_AUTH = 'INIT_AUTH';
export const CLOSE_AUTH = 'CLOSE_AUTH';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOG_OUT = 'LOG_OUT';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
