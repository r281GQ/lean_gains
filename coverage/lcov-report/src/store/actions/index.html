<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/store/actions</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/store/actions
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>45/45</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>45/45</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="app_actions.js"><a href="app_actions.js.html">app_actions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="21" class="abs high">21/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="21" class="abs high">21/21</td>
	</tr>

<tr>
	<td class="file high" data-value="auth_actions.js"><a href="auth_actions.js.html">auth_actions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="daily_logs_actions.js"><a href="daily_logs_actions.js.html">daily_logs_actions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

<tr>
	<td class="file high" data-value="user_details_actions.js"><a href="user_details_actions.js.html">user_details_actions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="workout_logs_actions.js"><a href="workout_logs_actions.js.html">workout_logs_actions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
