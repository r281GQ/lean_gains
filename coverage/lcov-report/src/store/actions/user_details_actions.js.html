<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/store/actions/user_details_actions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/store/actions</a> user_details_actions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9</td><td class="line-coverage quiet"><span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const WRITE_USER_DETAILS = 'WRITE_USER_DETAILS';
export const UNSET_PICTURE = 'UNSET_PICTURE';
export const WRITE_WORKOUT_TARGET = 'WRITE_WORKOUT_TARGET';
export const WRITE_WORKOUT_TARGETS = 'WRITE_WORKOUT_TARGETS';
export const DELETE_WORKOUT_TARGET = 'DELETE_WORKOUT_TARGET';
export const WRITE_KCAL_TARGETS = 'WRITE_KCAL_TARGETS';
export const WRITE_LATEST = 'WRITE_LATEST';
export const WRITE_KCAL_TARGET = 'WRITE_KCAL_TARGET';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Aug 22 2017 13:14:08 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
