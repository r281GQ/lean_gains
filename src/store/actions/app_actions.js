export const SELECT_WORKOUT_LOG_MONTH = 'SELECT_WORKOUT_LOG_MONTH';
export const OPEN_CONFIRM_DELETE_MODAL = 'OPEN_CONFIRM_DELETE_MODAL';
export const CLOSE_CONFIRM_DELETE_MODAL = 'CLOSE_CONFIRM_DELETE_MODAL';
export const SET_SELECTED_WORKOUT_LOG = 'SET_SELECTED_WORKOUT_LOG';
export const OPEN_SIDE_BAR = 'OPEN_SIDE_BAR';
export const CLOSE_SIDE_BAR = 'CLOSE_SIDE_BAR';
export const SELECT_DAILY_LOG_MONTH = 'SELECT_DAILY_LOG_MONTH';
export const SET_SELECTED_WORKOUT_TARGET = 'SET_SELECTED_WORKOUT_TARGET';
export const SET_SELECTED_DAILY_LOG = 'SET_SELECTED_DAILY_LOG';
export const INIT_API = 'INIT_API';
export const CLOSE_API = 'CLOSE_API';
export const OPEN_MESSAGE_BAR = 'OPEN_MESSAGE_BAR';
export const CLOSE_MESSAGE_BAR = 'CLOSE_MESSAGE_BAR';
export const SET_MESSAGE = 'SET_MESSAGE';
export const INIT_FETCH = 'INIT_FETCH';
export const CLOSE_FETCH = 'CLOSE_FETCH';
export const SET_CALORIE_LOG_DAY = 'SET_CALORIE_LOG_DAY';
export const OPEN_CONSENT_MODAL = 'OPEN_CONSENT_MODAL';
export const CLOSE_CONSENT_MODAL = 'CLOSE_CONSENT_MODAL';
export const ADD_CONSENT = 'ADD_CONSENT';
export const REMOVE_CONSENT = 'REMOVE_CONSENT';
export const UNSET_PENDING_CALORIE_LOG_DAY = 'UNSET_PENDING_CALORIE_LOG_DAY';
