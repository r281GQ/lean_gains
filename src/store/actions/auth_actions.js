export const INIT_AUTH = 'INIT_AUTH';
export const CLOSE_AUTH = 'CLOSE_AUTH';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOG_OUT = 'LOG_OUT';
