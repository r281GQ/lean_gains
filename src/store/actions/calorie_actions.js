export const WRITE_CALORIE_LOG = 'WRITE_CALORIE_LOG';
