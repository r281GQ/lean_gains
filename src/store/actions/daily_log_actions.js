export const WRITE_DAILY_LOG = 'WRITE_DAILY_LOG';
export const WRITE_DAILY_LOGS = 'WRITE_DAILY_LOGS';
export const WRITE_DAILY_LOG_DATES = 'WRITE_DAILY_LOG_DATES';
export const DELETE_DAILY_LOG = 'DELETE_DAILY_LOG';
