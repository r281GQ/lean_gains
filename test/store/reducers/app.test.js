import chai from 'chai';

import reducer from './../../../src/store/reducers/app';
import * as app from './../../../src/store/actions/app_actions';


describe('app reducer', () => {
  it('should set isLoading to TRUE on INIT_API', () => {
    const nextState = reducer(undefined, { type: app.INIT_API });
    expect(nextState).toMatchSnapshot();
  });

  it('should set isLoading to FALSE on CLOSE_API', () => {
    const nextState = reducer(reducer(undefined, { type: app.INIT_API }), {
      type: app.CLOSE_API
    });
    expect(nextState).toMatchSnapshot();
  });

  it('should set isSideBarOpen to TRUE on OPEN_SIDE_BAR', () => {
    const nextState = reducer(undefined, { type: app.OPEN_SIDE_BAR });
    expect(nextState).toMatchSnapshot();
  });

  it('should set isSideBarOpen to FALSE on CLOSE_SIDE_BAR', () => {
    const nextState = reducer(reducer(undefined, { type: app.OPEN_SIDE_BAR }), {
      type: app.CLOSE_SIDE_BAR
    });
    expect(nextState).toMatchSnapshot();
  });

  it('should set isConfirmDeleteModalOpen to TRUE on OPEN_CONFIRM_DELETE_MODAL', () => {
    const nextState = reducer(undefined, {
      type: app.OPEN_CONFIRM_DELETE_MODAL
    });
    expect(nextState).toMatchSnapshot();
  });

  it('should set isConfirmDeleteModalOpen to FALSE on CLOSE_CONFIRM_DELETE_MODAL', () => {
    const nextState = reducer(
      reducer(undefined, { type: app.OPEN_CONFIRM_DELETE_MODAL }),
      {
        type: app.CLOSE_CONFIRM_DELETE_MODAL
      }
    );
    expect(nextState).toMatchSnapshot();
  });

  it('should set selected daily log _id', () => {
    const nextState = reducer(undefined, {
      type: app.SET_SELECTED_DAILY_LOG,
      payload: 'randomId'
    });
    chai.expect(nextState.get('selectedDailyLog')).to.equal('randomId');
  });

  it('should set selected workout log _id', () => {
    const nextState = reducer(undefined, {
      type: app.SET_SELECTED_WORKOUT_LOG,
      payload: 'randomId'
    });
    chai.expect(nextState.get('selectedWorkoutLog')).to.equal('randomId');
  });

  it('should set selected workout target _id', () => {
    const nextState = reducer(undefined, {
      type: app.SET_SELECTED_WORKOUT_TARGET,
      payload: 'randomId'
    });
    chai.expect(nextState.get('selectedWorkoutTarget')).to.equal('randomId');
  });

  it('should set selected month for daily log', () => {
    const nextState = reducer(undefined, {
      type: app.SELECT_DAILY_LOG_MONTH,
      payload: '05-2016'
    });
    chai.expect(nextState.get('selectedMonthForDailyLogs')).to.equal('05-2016');
  });

  it('should set selected month for workout log', () => {
    const nextState = reducer(undefined, {
      type: app.SELECT_WORKOUT_LOG_MONTH,
      payload: '05-2016'
    });
    chai.expect(nextState.get('selectedMonthForWorkoutLogs')).to.equal('05-2016');
  });
});
