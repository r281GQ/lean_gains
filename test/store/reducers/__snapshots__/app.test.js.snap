// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app reducer should set isConfirmDeleteModalOpen to FALSE on CLOSE_CONFIRM_DELETE_MODAL 1`] = `
Immutable.Map {
  isConfirmDeleteModalOpen: false,
  isSideBarOpen: false,
  isLoading: false,
  isMessageBarOpen: false,
  isFetching: false,
  selectedDayCalorieLog: 2017-08-21T20:27:24.101Z,
  message: "",
  isConsentModalOpen: false,
}
`;

exports[`app reducer should set isConfirmDeleteModalOpen to TRUE on OPEN_CONFIRM_DELETE_MODAL 1`] = `
Immutable.Map {
  isConfirmDeleteModalOpen: true,
  isSideBarOpen: false,
  isLoading: false,
  isMessageBarOpen: false,
  isFetching: false,
  selectedDayCalorieLog: 2017-08-21T20:27:24.101Z,
  message: "",
  isConsentModalOpen: false,
}
`;

exports[`app reducer should set isLoading to FALSE on CLOSE_API 1`] = `
Immutable.Map {
  isConfirmDeleteModalOpen: false,
  isSideBarOpen: false,
  isLoading: false,
  isMessageBarOpen: false,
  isFetching: false,
  selectedDayCalorieLog: 2017-08-21T20:27:24.101Z,
  message: "",
  isConsentModalOpen: false,
}
`;

exports[`app reducer should set isLoading to TRUE on INIT_API 1`] = `
Immutable.Map {
  isConfirmDeleteModalOpen: false,
  isSideBarOpen: false,
  isLoading: true,
  isMessageBarOpen: false,
  isFetching: false,
  selectedDayCalorieLog: 2017-08-21T20:27:24.101Z,
  message: "",
  isConsentModalOpen: false,
}
`;

exports[`app reducer should set isSideBarOpen to FALSE on CLOSE_SIDE_BAR 1`] = `
Immutable.Map {
  isConfirmDeleteModalOpen: false,
  isSideBarOpen: false,
  isLoading: false,
  isMessageBarOpen: false,
  isFetching: false,
  selectedDayCalorieLog: 2017-08-21T20:27:24.101Z,
  message: "",
  isConsentModalOpen: false,
}
`;

exports[`app reducer should set isSideBarOpen to TRUE on OPEN_SIDE_BAR 1`] = `
Immutable.Map {
  isConfirmDeleteModalOpen: false,
  isSideBarOpen: true,
  isLoading: false,
  isMessageBarOpen: false,
  isFetching: false,
  selectedDayCalorieLog: 2017-08-21T20:27:24.101Z,
  message: "",
  isConsentModalOpen: false,
}
`;
